src/lib/users/dialogs/inst_user_first.rb

```diff
@@ -0,0 +1,661 @@
+# Copyright (c) 2006-2012 Novell, Inc. All Rights Reserved.
+#
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail, you may find
+# current contact information at www.novell.com.
+# ------------------------------------------------------------------------------
+
+# File:	clients/inst_user_first.ycp
+# Package:	Configuration of users and groups
+# Summary:	Dialog for creating the first user during installation
+# Authors:     Jiri Suchomel <jsuchome@suse.cz>
+#
+# $Id$
+require "yast"
+require "ui/event_dispatcher"
+require "users/dialogs/users_to_import"
+require "users/ca_password_validator"
+require "users/local_password"
+
+module Yast
+  class InstUserFirstDialog
+    include ::UI::EventDispatcher
+    include Yast::I18n
+    include Yast::UIShortcuts
+
+    def run
+      create_dialog
+      begin
+        event_loop
+      ensure
+        close_dialog
+      end
+    end
+
+    def initialize
+      import_yast_modules
+      textdomain "users"
+
+      @text_mode = UI.TextMode
+
+      # Widgets to enable/disable depending on the selected action
+      # (the first one receives the initial focus if applicable)
+      @widgets = {
+        new_user: [:full_name, :username, :pw1, :pw2, :root_pw, :root_mail, :autologin],
+        import: [:choose_users, :import_qty_label],
+        skip: []
+      }
+
+      # full info about imported users
+      @importable_users = {}
+      # user names of imported users
+      @importable_usernames = []
+      # names of imported users selected for writing
+      @usernames_to_import = []
+
+      # if importing users from different partition is possible
+      @import_available = UsersSimple.ImportAvailable
+
+      if @import_available
+        @importable_users = UsersSimple.GetImportedUsers("local")
+        @importable_usernames = @importable_users.keys
+
+        if @importable_usernames.empty?
+          Builtins.y2milestone("No users to import")
+          @import_available = false
+        end
+      end
+
+      # do not open package progress wizard window
+      @progress_orig = Progress.set(false)
+
+      @users = UsersSimple.GetUsers
+      @action = case @users.size
+      when 0
+        # New user is the default option
+        GetInstArgs.going_back ? :skip : :new_user
+      when 1
+        @users.first["__imported"] ? :import : :new_user
+      else
+        :import
+      end
```

Comment:
Reshuffled a little bit. Still, it's old code, so I consider it improved enough. :-)